analytical_engine/core/fragment/dynamic_fragment_view.h

```diff
@@ -0,0 +1,263 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ANALYTICAL_ENGINE_CORE_FRAGMENT_DYNAMIC_FRAGMENT_VIEW_H_
+#define ANALYTICAL_ENGINE_CORE_FRAGMENT_DYNAMIC_FRAGMENT_VIEW_H_
+
+#ifdef EXPERIMENTAL_ON
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "core/fragment/dynamic_fragment.h"
+
+namespace gs {
+
+/**
+ * @brief A wrapper class of DynamicFragment to behaves as a view.
+ *
+ */
+class DynamicFragmentView : public DynamicFragment {
+ public:
+  using fragment_t = DynamicFragment;
+
+  DynamicFragmentView(fragment_t* frag, const std::string& view_type)
+      : fragment_(frag), view_type_(view_type) {}
+
```

Comment:
add a destructor to release the share_ptr.

Reply:
frag is a regular ptr, we do not need to release

Reply:
It's such a strange design pattern that you have supplied a factory method for creating a `DynamicFragmentView` instance. Generally, static methods for creating instances just when there is a lot of arguments are required to assemble objects. In another, usually, a method named `Init` when the method is time-consuming when it is invoked. If you persist to use the `Init`, you may make the constructor to private to prevent miss-using from users.